### ISSUE TITLE
List Widget with nested fields creates nonfunctional UI in CMS

### ISSUE THREAD

**Issue by jvoros** [netlify/netlify-cms#1607]
**- Do you want to request a *feature* or report a *bug*?**
Bug

**- What is the current behavior?**
After configuring a collection based on the YAML in the docs, with a list widget and nested fields, the UI renders a form with nonfunctional fields. The labels do not render, I am only able to enter a single character in the form fields.

![Imgur](https://i.imgur.com/wBm4IhT.png)

**- If the current behavior is a bug, please provide the steps to reproduce.**
Configure a list widget item with a single sub-field.

**- What is the expected behavior?**
The UI should render labeled form fields that accept more input than a single character.

**- Please mention your versions where applicable.**

Netlify CMS version: 2.0.9
Browser version: Chrome 67
Operating System: mac OS

**- Please link or paste your `config.yml` below if applicable.**

```
backend:
  name: github
  repo: aliemteam/aliemcards_cards
  branch: netlifycms
publish_mode: editorial_workflow
media_folder: "cards/media"
public_folder: "media"
slug:
  encoding: "ascii"
  clean_accents: true
collections:
  - name: "cards" # Used in routes, e.g., /admin/collections/blog
    label: "Cards" # Used in the UI
    folder: "/cards" # The path to the folder where the documents are stored
    create: true # Allow users to create new documents in this collection
    slug: "{{slug}}" # Filename template, e.g., YYYY-MM-DD-title.md
    fields: # The fields for each document, usually in front matter
      - {label: "Title", name: "title", widget: "string"}
      - {label: "Created", name: "created", widget: "string"}
      - {label: "Updated", name: "updated", widget: "string", required: false}
      - {label: "Categories", name: "categories", widget: "list"}
      - label: "Card Authors"
        name: "authors"
        widget: "list"
        field:
          - { label: "Name", name: "name", widget: "string" }
      - {label: "Body", name: "body", widget: "markdown"}
```

**Comment by jvoros** [comment]
Seems related to #1139 and #1574.

Making `field` plural to `fields` causes error: `TypeError: e.get is not a function` triggered by `immutable.js` according to console.

**Comment by jvoros** [comment]
A workaround for my use case would be to allow custom delimiters for the list widget.

**Comment by jvoros** [comment]
Hoping #1395 will fix this.

**Comment by kamontat** [comment]
Same issue on gatsby

**Comment by erquhart** [comment]
This was addressed in 2.1.2.
